queries/java.scm

```diff
@@ -0,0 +1,135 @@
+;; Generated by the following command:
+;; > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-java/master/src/node-types.json | jq '[.[] | select(.type == "statement" or .type == "declaration") | .subtypes[].type]'
+[
+    (annotation_type_declaration)
+    (class_declaration)
+    (enum_declaration)
+    (import_declaration)
+    (interface_declaration)
+    (module_declaration)
+    (package_declaration)
+    (assert_statement)
+    (break_statement)
+    (continue_statement)
+    (declaration)
+    (do_statement)
+    (enhanced_for_statement)
+    (expression_statement)
+    (for_statement)
+    (if_statement)
+    (labeled_statement)
+    (local_variable_declaration)
+    (return_statement)
+    (switch_expression)
+    (synchronized_statement)
+    (throw_statement)
+    (try_statement)
+    (try_with_resources_statement)
+    (while_statement)
+    (yield_statement)
+
+    ;; exceptions
+    ;; ";",
+    ;; "block",
+    (method_declaration)
+    (constructor_declaration)
+    (field_declaration)
+] @statement
+
+(class_declaration
+    name: (_) @className
+) @class @className.domain
+
+(method_declaration
+    (identifier) @functionName
+) @namedFunction @functionName.domain
+(constructor_declaration
+    (identifier) @functionName
+) @namedFunction @functionName.domain
+
+(lambda_expression) @anonymousFunction
+
+(
+    (if_statement) @ifStatement
+    (#not-parent-type? @ifStatement "if_statement")
```

Comment:
This doesn't match what was there before. I'm assuming this is an improvement, but worth adding a test to capture the new behaviour if so

Reply:
done